ms-engine/src/main/java/net/trajano/ms/engine/internal/VertxRoutingContextHttpHeaders.java

```diff
@@ -0,0 +1,134 @@
+package net.trajano.ms.engine.internal;
+
+import java.time.Instant;
+import java.time.format.DateTimeFormatter;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import javax.ws.rs.core.Cookie;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedMap;
+
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * There's a bug in RestEasy's implementation of HTTP headers in that it is not
+ * case insensitive when looking up keys.
+ *
+ * @author Archimedes Trajano
+ */
+public class VertxRoutingContextHttpHeaders implements
+    HttpHeaders {
+
+    private final RoutingContext context;
+
+    private final MediaType mediaType;
+
+    public VertxRoutingContextHttpHeaders(final RoutingContext context) {
+
+        this.context = context;
+
+        final String obj = getHeaderString(CONTENT_TYPE);
+        if (obj == null) {
+            mediaType = MediaType.APPLICATION_OCTET_STREAM_TYPE;
+        } else {
+            mediaType = MediaType.valueOf(obj);
+        }
+    }
+
+    @Override
+    public List<Locale> getAcceptableLanguages() {
+
+        return context.acceptableLanguages().stream().map(r -> {
+            if (r.subtagCount() == 0) {
+                return new Locale(r.tag());
+            } else if (r.subtagCount() == 1) {
+                return new Locale(r.tag(), r.subtag());
+            } else {
+                return new Locale(r.tag(), r.subtag(), r.subtag(2));
+            }
+        }).collect(Collectors.toList());
+
+    }
+
+    @Override
+    public List<MediaType> getAcceptableMediaTypes() {
+
+        final String acceptableContentType = context.getAcceptableContentType();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "mediaType" which hides the field declared at line 30. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)